parley/src/layout/line/greedy.rs

```diff
@@ -252,50 +253,29 @@ impl<'a, B: Brush> BreakLines<'a, B> {
 
                         // Retrieve metadata about the cluster
                         let is_ligature_continuation = cluster.is_ligature_continuation();
-                        let is_space = cluster.info().whitespace().is_space_or_nbsp();
+                        let whitespace = cluster.info().whitespace();
+                        let is_newline = whitespace == Whitespace::Newline;
+                        let is_space = whitespace.is_space_or_nbsp();
                         let boundary = cluster.info().boundary();
```

Comment:
The full explanation is at https://github.com/linebender/parley/issues/132#issuecomment-2408754466

tldr; Unicode LBA specifies a break _after_ the newline and a trailing newline doesn’t have a following cluster so we lose that information. 

Re check order: good question. I’m going to assume that a newline sequence is not marked as a break opportunity because the code appears to work as is :)